Panic in Reconciler

**Describe the bug**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8dcb18]

goroutine 63 [running]:
github.com/coinbase/rosetta-sdk-go/reconciler.(*Reconciler).reconcileInactiveAccounts(0xc0073e2210, 0xbe28a0, 0xc005b5b080, 0x0, 0x0)
        /home/ec2-user/go/src/github.com/coinbase/rosetta-sdk-go/reconciler/reconciler.go:644 +0xe8
github.com/coinbase/rosetta-sdk-go/reconciler.(*Reconciler).Reconcile.func2(0xc008ff6768, 0x0)
        /home/ec2-user/go/src/github.com/coinbase/rosetta-sdk-go/reconciler/reconciler.go:707 +0x3f
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0063b2270, 0xc0067e5bc0)
        /home/ec2-user/go/src/golang.org/x/sync/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/ec2-user/go/src/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
```

**To Reproduce**
1) Sync some blocks to a persistent datadir using `rosetta-cli check`
2) Halt syncing
3) Restart `check` from the previous datadir

--- a/reconciler/reconciler.go
+++ b/reconciler/reconciler.go
@@ -641,10 +641,20 @@ func (r *Reconciler) reconcileInactiveAccounts(
 		}
 
 		r.inactiveQueueMutex.Lock()
+		if len(r.inactiveQueue) == 0 {
+			r.inactiveQueueMutex.Unlock()
+			if r.debugLogging {
+				log.Println(
+					"no accounts ready for inactive reconciliation (0 accounts in queue)",
+				)
+			}
+			time.Sleep(inactiveReconciliationSleep)
+			continue
+		}
+
 		nextValidIndex := r.inactiveQueue[0].LastCheck.Index + r.inactiveFrequency
-		if len(r.inactiveQueue) > 0 &&
-			(r.inactiveQueue[0].LastCheck == nil || // block is set to nil when loaded from previous run
-				nextValidIndex <= head.Index) {
+		if r.inactiveQueue[0].LastCheck == nil || // block is set to nil when loaded from previous run
+			nextValidIndex <= head.Index {
 			nextAcct := r.inactiveQueue[0]
 			r.inactiveQueue = r.inactiveQueue[1:]
 			r.inactiveQueueMutex.Unlock()

--- a/reconciler/reconciler.go
+++ b/reconciler/reconciler.go
@@ -652,10 +652,13 @@ func (r *Reconciler) reconcileInactiveAccounts(
 			continue
 		}
 
-		nextValidIndex := r.inactiveQueue[0].LastCheck.Index + r.inactiveFrequency
-		if r.inactiveQueue[0].LastCheck == nil || // block is set to nil when loaded from previous run
-			nextValidIndex <= head.Index {
-			nextAcct := r.inactiveQueue[0]
+		nextAcct := r.inactiveQueue[0]
+		nextValidIndex := int64(-1)
+		if nextAcct.LastCheck != nil { // block is set to nil when loaded from previous run
+			nextValidIndex = nextAcct.LastCheck.Index + r.inactiveFrequency
+		}
+
+		if nextValidIndex <= head.Index {
 			r.inactiveQueue = r.inactiveQueue[1:]
 			r.inactiveQueueMutex.Unlock()
 

